File: ProjectApplication/lib/model/Member.dart
class Member {
  final int? memberId;
  final String? firstName;
  final String? lastName;
  final String? birthDate;
  final String? gender;
  final String? profileImage;
  final String? email;
  final String? password;
  final String? phoneNumber;

  Member({
    this.memberId,
    this.firstName,
    this.lastName,
    this.birthDate,
    this.gender,
    this.profileImage,
    this.email,
    this.password,
    this.phoneNumber,
  });

  factory Member.fromMemberJson(Map<String, dynamic> json) {
    return Member(
      memberId: json['memberId'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      birthDate: json['birthDate'],
      gender: json['gender'],
      profileImage: json['profileImage'],
      email: json['email'],
      password: json['password'],
      phoneNumber: json['phoneNumber'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'memberId': memberId,
      'firstName': firstName,
      'lastName': lastName,
      'birthDate': birthDate,
      'gender': gender,
      'profileImage': profileImage,
      'email': email,
      'password': password,
      'phoneNumber': phoneNumber,
    };
  }

  Member copyWith({
    int? memberId,
    String? firstName,
    String? lastName,
    String? birthDate,
    String? gender,
    String? profileImage,
    String? email,
    String? password,
    String? phoneNumber,
  }) {
    return Member(
      memberId: memberId ?? this.memberId,
      firstName: firstName ?? this.firstName,
      lastName: lastName ?? this.lastName,
      birthDate: birthDate ?? this.birthDate,
      gender: gender ?? this.gender,
      profileImage: profileImage ?? this.profileImage,
      email: email ?? this.email,
      password: password ?? this.password,
      phoneNumber: phoneNumber ?? this.phoneNumber,
    );
  }
}

File: Restaurant/src/main/java/com/finalproject/Restaurant/model/Member.java
package com.finalproject.Restaurant.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.*;

@Entity
@Table(name = "member")
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    private Integer memberId;

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @Temporal(TemporalType.DATE)
    @Column(name = "birth_date")
    private Date birthDate;


    @Column(name = "gender", length = 10)
    private String gender;

    @Column(name = "profile_image")
    private String profileImage;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "password", nullable = false, length = 100)
    private String password;

    @Column(name = "phone_number", length = 13)
    private String phoneNumber;

}
